### ISSUE TITLE
I will fix this App.

### ISSUE THREAD

**Issue by losh11** [hivewallet/hive-ios#52]
I know I am really busy, but I don't want this to go un-maintained.
Currently this is the only iOS app which works with Litecoin and that is important. And because of that, I will contribute to this project in order to get this back fully working on the iOS App Store.

**Results to expect:-**
- Everything that works on hive-js should work on this (incl fees)
- All crashes to do with barcodes, cameras.
- Crashes to do with addresses and validity.
- And some more (if and when discovered)...

But this won't take place until some conditions are met, they aren't thermo-nuclear, just more of a little demand. Like git permissions and other git stuff.

**Conditions needing to be met:-**
- Add the user 'losh11' (me) to a team which can push to this repository.
- Create a new branch which has the pull request (regarding the phonegap updates) is committed.

And all of this will be completed by the end of the week (Sunday evening). I can't do debugging, but hopefully everything works all right without much problems, that's why I recommended creating a new working branch. If you want to remove me from push requests after I have done this, you can. I will not ask for any money whatsoever, but don't mind donations (you will somehow find my LTC address online - maybe LTCtalk). I won't advertise myself with the app, not officially, maybe on Twitter (personal account).

The faster the conditions are met, the quicker I can get all this work finished.

**Comment by losh11** [comment]
@w-hive

**Comment by ghost** [comment]
Wow Losh, that's extraordinarily generous of you. But why don't you fork the project (and the hive-js project) and we will merge when you're done? In fact that's how we even worked internally. Then you will have the full ability to create branches, etc.

**Comment by losh11** [comment]
Alright, I'll do it on a fork. How will you review it though, I might need
you to send out a couple test flight invitations for people to test the
prerelease.

I know you're not a iOS/Mac guy, so do you have a Mac with Xcode and other
development tools? I'm presuming that you will be the one uploading to
iTunes Connect, need any help there?

**Comment by ghost** [comment]
Sure, that should be no problem about Testflight invitations. I don't have a Mac that would really be useful for any of the development stuff here (it's an old one stuck on an old OS). Likewise, I've not _personally_ uploaded to iTunes before, but I of course have the credentials (and have tweaked the description, etc). _Maybe_ we can prod Kuba if we need any assistance with that. Maybe. :)

Regarding review, I'll ask if we can get a little bit of @weilu 's time on it. Otherwise maybe @tgerring would be willing. If not, we'll just have to wing it with a more public review.

**Comment by ghost** [comment]
Hey Losh, how are you doing with all of this?

**Comment by losh11** [comment]
Hey, I think I pretty much done, only like 30 mins of work left with a little testing.
The problem with all this is how I run the server. I am pretty confused since tutorials on Hive-JS don't seem to be in any correct order.

Where do I enter the stuff that is inside the 'profit section'? The bit that goes like this:

> DB_HOST=127.0.0.1 DB_PORT=5984 DB_USER=admin DB_PASSWORD=password COOKIE_SALT=secret PROXY_URL=https://hive-proxy.herokuapp.com?url= npm run dev
> open http://localhost:8080

It's probably just that I'm a bit stupid.

**Comment by tgerring** [comment]
`DB_HOST=127.0.0.1 DB_PORT=5984 DB_USER=admin DB_PASSWORD=password COOKIE_SALT=secret PROXY_URL=https://hive-proxy.herokuapp.com?url= npm run dev`

This is setting DB_HOST, DB_PORT, DB_USER, DB_PASSWORD, COOKIE_SALE, and PROXY_URL environment variables for the `npm run dev` command

`open http://localhost:8080` launches the site in your browser. On production, this is unnecessary because a webserver (like nginx) probably sits between the script and the user.

**Comment by ghost** [comment]
Any updates Losh? Was Taylor's information helpful?

**Comment by losh11** [comment]
Yes, Pretty much everything done. I just want to move the toolbar to the
bottom of the screen like most other iOS apps.

Created a new Cordova project right from hive-is source.

Decided to create a script that could compile, run sim, run on device and
deploy instead of using commands relying on npm packages which don't always
work.

Look out, there should be a new pull request any day...

**Comment by ghost** [comment]
So do I understand correctly then, that you're basically going to keep a hive-js repo, with parameters to specifically make the iOS build, and are not using any of the work we already put into the iOS repo (including things like toolbar positioning, camera plugin, etc)? Yours might actually be a smarter way to do it, but just wondering.

By the way Losh, is there an email address that we can use to reach you?

**Comment by losh11** [comment]
Redoing that all again on top of the hivejs repo.

**Comment by tstock8** [comment]
Did you fix this app?

**Comment by tstock8** [comment]
I still have bitcoins in Hive and can't retrieve them.

**Comment by losh11** [comment]
This is how you retrieve your Hive coins (sorry stolen from an email)!

Go here: https://dcpos.github.io/bip39/
1. Set number of words to 12, and enter the seed into the mnemonic field.
2. Enter the password if you had one
3. Set it to BTC, and click on the BIP32 tab (NOT BIP39)
4. The BIP32 derivation path needs to say m/0'/0
5. Take each private key and import into another Bitcoin Wallet (such as
   Electrum/Breadwallet/mycelium/copay)
